src/main/java/org/openstreetmap/atlas/checks/validation/areas/PedestrianAreaOverlappingEdgeCheck.java

```diff
@@ -0,0 +1,242 @@
+        final List<Segment> segments = area.asPolygon().segments();
+        // Get all start and end locations of all segments in the area
+        final List<Location> locationsInSegments = Segment.asList(segments);
+        // Filter and collect all valid intersecting edges
+        final Set<Edge> filteredIntersectingEdges = Iterables
+                .stream(area.getAtlas().edgesIntersecting(area.asPolygon()))
+                .filter(edge -> this.isValidIntersectingEdge(edge, area)).collectToSet();
+        final Set<AtlasObject> overlappingEdges = new HashSet<>();
+        final Polygon areaPolygon = area.asPolygon();
+        for (final Edge edge : filteredIntersectingEdges)
+        {
+            final Location edgeStartLocation = edge.start().getLocation();
+            final Location edgeEndLocation = edge.end().getLocation();
+            // Checks if intersection is at start of the edge
+            final boolean intersectsAtStart = locationsInSegments.contains(edgeStartLocation);
+            // Checks if intersection is at end of the edge
+            final boolean intersectsAtEnd = locationsInSegments.contains(edgeEndLocation);
+            // Filter the segments with the intersecting points
+            final Set<Segment> filteredSegments = segments.stream()
+                    .filter(segment -> edgeStartLocation.equals(segment.start())
+                            || edgeStartLocation.equals(segment.end())
+                            || edgeEndLocation.equals(segment.start())
+                            || edgeEndLocation.equals(segment.end()))
+                    .collect(Collectors.toSet());
+            // If both ends of the edge intersects the area, check if it is properly snapped by
+            // checking
+            // if both the intersecting points are on the same segment.
+            // If none of the filtered segments have both the start and end nodes, then add the
+            // edge and all its connected edges that are within the area to flag them.
+            if ((intersectsAtStart && intersectsAtEnd && this
+                    .isNotSnappedToSegments(filteredSegments, edgeStartLocation, edgeEndLocation))
+                    ||
+                    // If any one of the end of the connected edge is fully enclosed within the
+                    // area,
+                    // flag the edge and all its connected edges that are within the area.
+                    (intersectsAtStart && !intersectsAtEnd
+                            && areaPolygon.fullyGeometricallyEncloses(edgeEndLocation))
+                    || (intersectsAtEnd && !intersectsAtStart
+                            && areaPolygon.fullyGeometricallyEncloses(edgeStartLocation)))
+            {
+                // Collect all connected edges that are within the pedestrian area
+                edge.connectedEdges().stream()
+                        .filter(connectedEdge -> connectedEdge.isMasterEdge() && areaPolygon
```

Comment:
Note: There might be updates to the filter for the connected edges based on analysis results.